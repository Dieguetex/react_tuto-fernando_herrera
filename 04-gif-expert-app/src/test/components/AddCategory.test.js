import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Pruebas en <AddCategoy />', () => {
	const setCategories = () => {};
	const wrapper = shallow(<AddCategory setCategories={setCategories} />);

	test('Debe mostrarse correctamente', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('should Debe de cambiar la caja de texto', () => {
		const input = wrapper.find('input');
		const value = 'hola mundo';

		input.simulate('change', { target: { value } });

		expect(wrapper.find('p').text().trim()).toBe(value);
	});
});
