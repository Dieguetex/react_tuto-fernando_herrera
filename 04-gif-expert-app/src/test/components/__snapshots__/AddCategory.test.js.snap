// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategoy /> Debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Search gifs"
    type="text"
    value=""
  />
</form>
`;
